Adds support for boost 1.59.0

toonetown opened the pull request:
Based upon patch at https://github.com/mutanabbi/chewy-cmake-rep/commit/ca7221f916ce631dd14b3be81e2c8d6b4d57659e.  Fixes #10.  See also #13.

It is based off the latest master branch, and only contains the changes needed for making the code support boost 1.59.0.  

These changes will work for boost 1.59.0 - but not previous versions, since some variables and function signatures changed.  Probably the best solution would be to do some sort of condition, but I am not familiar enough with the boost::test library to suggest what condition to key off of (I literally just took the changes from the above patch and based them off the master branch)

toonetown commented:
It looks to me like the appveyor failure is not related to this change...is there something I'm doing wrong?

zaufi commented on boost/teamcity_boost.cpp:
```diff
@@ -17,21 +17,33 @@
 
 namespace jetbrains { namespace teamcity {
+const std::string ASSERT_CTX = "Assertion occurred in a following context:";
+const std::string FAILURE_CTX = "Assertion occurred in a following context:";
```
should be "Failure ..." text here... copy'n'paste typo

zaufi commented on boost/teamcity_boost.cpp:
```diff
@@ -17,21 +17,33 @@
     TeamcityMessages messages;
     std::string currentDetails;
+    std::string currentContextDetails;
+    std::string currentTestName;
```
you don't need this actually. it is always possible to get a current test name. see my fork of this...

zaufi commented on boost/teamcity_boost.cpp:
```diff
@@ -71,7 +94,7 @@ struct TeamcityFormatterRegistrar {
+BOOST_GLOBAL_FIXTURE(TeamcityFormatterRegistrar);
 
 // Formatter implementation
 static std::string toString(boost::unit_test::const_string bstr) {
```
why not to move this to anonymous namespace instead of C style `static` keyword...

toonetown replied:
Just wanted to keep the changes to a minimum for a single patch/feature.  IMO, things like moving functions around should be done in a separate patch, when possible.

zaufi replied:
good to you ;-) I did a lot of changes and now it is not possible to merge them in that way -- it is too late...

toonetown replied:
I did try and go through your fork and add them to this patch (the ones that appeared to be relevant...like the conditional stuff and the current test name).  Maybe there are other parts to it that I have missed?

zaufi commented:
>  Probably the best solution would be to do some sort of condition, 

yes, it is. I've done this in my fork...

toonetown commented:
Yeah - I didn't look at the whole fork - just that one commit that was mentioned in Issue #13...I didn't know if there was more work going on (and the lack of 1.59.0 support was biting me) so I figured I'd work up a full PR in the meantime.  Looks like the stuff that you have going on in your fork is probably a better long-term solution, and is probably preferable to this PR for inclusion.

One note (that I didn't make clear) is that I actually *did* update this patch with a conditional (I base it off of BOOST_VERSION >= 105900) - and from my testing works with Boost 1.58 and 1.59 (I haven't tested other versions - but I'm guessing the they would work also).

Again - feel free to close the PR and merge in a more "official" one if needed.  I just thought I'd contribute back what I had found during my course of working with it.

zaufi commented:
Yes, my fork works (and tested) for boost 1.59 and prior. And yes, I also intended to do some more improvements. But my PR was declined because of formatting...

toonetown commented:
Just updated to remove the storage of the test name - however, in the fork at https://github.com/zaufi/teamcity-cpp/commit/d2ec061ca9b85252be6475a080d81b3db8127783, it uses full_name() which is not available in 1.58 and previous (it's p_name...and not a function) - so I added that to the conditional as well.

zaufi commented:
>  it uses full_name() which is not available in 1.58 and previous (it's p_name...and not a function) - so I added that to the conditional as well.

Yeah, it is in `#if` condition at that patch...

shalupov commented:
Thank you! Thank you, @zaufi, for original pull request!

I'll review and merge as soon as I set up old/new boost configurations on Travis.
(most likely on this weekends)
